src/transforms/rename.jl

```diff
@@ -0,0 +1,37 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+  Rename(:col₁ => :newcol₁, :col₂ => :newcol₂, ..., :col₁ => :newcolₙ)
+
+The transform that renames `col₁` to `newcol₁`, `col₂` to `newcol₂`, ...
+"""
+struct Rename <: Stateless
+  names::Dict{Symbol,Symbol}
+end
+
+function apply(transform::Rename, table)
+  oldnames = Tables.columnnames(table)
+  newnames = map(oldnames) do oldname
+    if oldname in keys(transform.names) 
+      transform.names[oldname]
+    else
+      oldname
+    end
+  end
+  acols = [i for i in Tables.columns(table)]
```

Comment:
I think this is not fully compatible with the Tables.jl interface. The result of `Tables.column` is not necessarily iterable. We need to use `Tables.getcolumn(cols, name)` to access a specific column.

Reply:
TIL. I noticed that it was not working with DataFrames, but it was also not working with JuliaDB, and I saw other `Tables.column` instances throughout the codebase so I thought it was fine for now. Turns out JuliaDB does not fully implement Tables.jl, and your other usages were not iterating.